### ISSUE TITLE
Why is validation/test set needed for unsupervised setting?

### ISSUE THREAD

**Issue by vinaysetty** [williamleif/GraphSAGE#56]
Hi,

I am trying to use graphsage to train embeddings in completely unsupervised way. But when I prepare the data in -G.json format, if I have 0 nodes with 'val' attribute set to True, then I get a crash when the model starts to train. What is the significance of validation in unsupervised setting? Ideally for unsupervised setting 'val' and 'test' attributes shouldn't be needed at all.

Also does it make sense to use the graphsage embeddings obtained in unsupervised way for some other task like link prediction?

Thanks.

**Comment by GamalC** [comment]
Thanks for this question. I am also in the same situation (assuming you mean 'val' and 'test' attributes).
I also intend to use the embeddings for link prediction; I can see no reason why they shouldn't be useful. I think that the task of node classification is quite tightly coupled in the code so simply using the unsupervised setting to get the embeddings is unexpectedly difficult.

**Comment by vinaysetty** [comment]
Yes I mean 'val' and 'test' attributes. I figured the crash can be fixed by simply setting 'val' attribute as 'True' for some random nodes.. I guess validation against a loss function in unsupervised setting also makes sense but when the goal is learn an embeddings for each every node in the network it is unclear the role of 'train' and 'val' nodes..
